eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/ContentTypeIdentifier.php

```diff
@@ -75,7 +76,11 @@ public function handle(
         $idList = [];
 
         foreach ($criterion->value as $identifier) {
-            $idList[] = $this->contentTypeHandler->loadByIdentifier($identifier)->id;
+            try {
+                $idList[] = $this->contentTypeHandler->loadByIdentifier($identifier)->id;
+            } catch (NotFoundException $e) {
+                // Skip non-existing content types
+            }
```

Comment:
What will happen if `$idList` will be empty?

Reply:
Good point, it can indeed fail. But then again, this code doesn't really check that `$criterion->value` is empty neither. I can simply have the following criterion:

```
$criteria[] = new Criterion\ContentTypeIdentifier([]);
```

and it will fail with the same error.

I can optionally return something in lieu of `return $query->expr->eq(0, 1);`, which would make the criterion not matching anything, which would be expected I guess.

Reply:
Hmm, by allowing this change we might want to do that, or ignore (do nothing) as is the case with one invalid _(ignored now with this change)_ and one valid _(still taken into account)_.

On the same line of topics this change opens up: Should we perhaps write some warning on invalid types being passed so people have a way of tracking down mistakes in code?

Reply:
> I can optionally return something in lieu of return $query->expr->eq(0, 1);, which would make the criterion not matching anything, which would be expected I guess.

Yes,  
`\eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriterionHandler\MatchNone::handle` might be inspiration 

> On the same line of topics this change opens up: Should we perhaps write some warning on invalid types being passed so people have a way of tracking down mistakes in code?

Or we can control this behaviour by adding `ignoreMissing` flag to `\eZ\Publish\API\Repository\Values\Content\Query\Criterion\ContentTypeIdentifier`, WDYT ?

Reply:
> Or we can control this behaviour by adding ignoreMissing flag to \eZ\Publish\API\Repository\Values\Content\Query\Criterion\ContentTypeIdentifier, WDYT ?

Could , but then I would expect it to be all over, and that won't fly.

Lets rather keep it simple+consistent and make sure this aligns with how `ContentTypeId` and other Id lookups work. Which is afaik what is being proposed here (ignore not found type by identifier lookups, match none of no one is found)